### PR TITLE
WCT website updates

### LINKED ISSUE #3661: WCT Redesign of About/Donation and Teams/Committees Pages
The WCT has completed some mockups of changes to the WCA website. Our goals were:

- to add donation information to the about page
- make the about page better showcase the organization
- put teams and committees on their own page with increased legibility
- reduce duplication between the about and contact pages for listing team and committee members
- use the contact form for most contact requests rather than the current contact page, which often results in emails going to the wrong place

So, we have made the following changes:
- new about page with donation information to replace the old one
- new teams and committees page that includes description, members, and contact information
- contact links now go to new contact form that automatically routes emails to the right place rather than making people read a list of emails

The mockups can be viewed at the links below. We would like some feedback from the WST, WFC, and the Board, but also from anybody else who has feedback on these ideas. Public commenting is enabled on all of these links, so if you have feedback about a specific part of the design, please feel free to leave a comment.

[New about page](https://drive.google.com/file/d/10tnEOr6XF6Z5jaZC8_yp6Upk3tfgDeA6/view) (the "Donate with PayPal" now says "Donate with Credit Card" and links to our Guidestar donation page)
[New teams and coommittees page](https://drive.google.com/file/d/1KCuZLrVRYdjic7y9cQ87sPmxllRJkD6S/view) (Board members and officers have been added to this page, please see the discussion below for screenshots)
[Teams and committees in the navbar menu](https://drive.google.com/file/d/17zagSmal0NctkKk92uW2QScj4jCCSVd6/view)

### PR DESCRIPTION
**Pull request by UnsolvedCypher**
Closes #3661

### CODE REVIEW

**Comment by AlbertoPdRF** [comment]
@thewca/board the mockups are in the issue #3661, can someone review?

File: WcaOnRails/config/locales/cs.yml [hunk 1]
```diff
@@ -2284,7 +2284,7 @@ cs:
         #original_hash: abd2828
         name: WCA Board of Directors
         #original_hash: d855b50
-        info_html: >-
+        description: >-
```
[the remark this answers is not in the document]
Comment (UnsolvedCypher) [reply WcaOnRails/config/locales/cs.yml]: This keeps the pattern of `name` and `description` that every other team has. I think it's ok not to have this *strong* since none of the other descriptions are. What do you think?

File: WcaOnRails/config/routes.rb [hunk 2]
```diff
@@ -31,6 +31,7 @@
   post 'admin/avatars' => 'admin/avatars#update_all'
 
   get 'competitions/mine' => 'competitions#my_competitions', as: :my_comps
+  get 'competitions/for_senior(/:user_id)' => 'competitions#for_senior', as: :competitions_for_senior
```
[the remark this answers is not in the document]
Comment (UnsolvedCypher) [reply WcaOnRails/config/routes.rb]: I actually have no idea what this is, perhaps it's from another branch. I'll get rid of that :)

File: WcaOnRails/config/locales/cs.yml (hunk 1 above)
[the remark this answers is not in the document]
Comment (jonatanklosko) [reply WcaOnRails/config/locales/cs.yml]: Sounds good to me, so there are two ways you can go about this:
1. Update only the `en.yml` changing the key name and removing the `strong` tag. This will require other translators to translate the description again (without the `strong` tag).
2. Update all files changing the key name and removing the `strong` tag. This saves our translators the work, but requires more work from you.
Whichever you prefer =)

[the remark this answers is not in the document]
Comment (UnsolvedCypher) [reply WcaOnRails/config/locales/cs.yml]: Ok, I'm happy to do a little more work =). It's my understanding that translation changes are automatically detected using the previous hash- do I need to update the hash somehow?

File: WcaOnRails/app/views/contacts/website.html.erb [hunk 3]
```diff
@@ -1,7 +1,9 @@
 <% provide(:title, t('.title')) %>
 <div class="container">
   <h1><%= yield(:title) %></h1>
-
+  <div class="alert alert-info">
+      Before making an inquiry, you may wish to take a look at our <%= link_to 'Frequently Asked Questions', '/faq' %>! If you wish to contact a specific committee or team, you can find their contact information on our <%= link_to 'about page', '/about' %>.
+  </div>
```
[the remark this answers is not in the document]
Comment (jonatanklosko) [reply WcaOnRails/app/views/contacts/website.html.erb]: I think in this case you can just hard code the link as an HTML tag, like here:
https://github.com/thewca/worldcubeassociation.org/blob/a78a9f119785c518c449861e77f2906040b71631/WcaOnRails/config/locales/en.yml#L1638
This is generally fine for short links that are constant (i.e. doesn't require any ID).

When doing this, just make sure to suffix the i18n key with `_html` (e.g. `faq_note_html`), so that it gets rendered as HTML, not escaped as a normal text.

File: WcaOnRails/config/locales/cs.yml (hunk 1 above)
[the remark this answers is not in the document]
Comment (jonatanklosko) [reply WcaOnRails/config/locales/cs.yml]: If you remove the strong tags from `en.yml`, so that the original text changes, then indeed it will be detected as outdated. But if you also remove `#original_hash` from the translation files, then it will fit the scenario "a new key as well as its translation has been added", thus will work as we want.
tl;dr; remove `#original_hash`es when going for `2.` =)

**Comment by AlbertoPdRF** [comment]
I think adding a "WCA Officers" team is a better long-term option. If the @thewca/software-team agrees, I can work on this in the next couple of days!

File: WcaOnRails/app/views/static_pages/teams_committees.html.erb [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+        <%= render "shared/user_avatar", user: user %>
+        <div class="team-member-name"><%= user.name %>
+          <br />
+          <span class="team-leader-subtext"><%= position %></span>
+        </div>
+      </div>
+  <% end %>
+
+  <% Team.all_ordered_by_english_name.each do |team| %>
+    <h3><%= team.name %> <%= team != Team.board ? "(" + team.acronym + ")" : "" %> <span class="badge team-email"><%= mail_to team.email %></span></h3>
```
[the remark this answers is not in the document]
Comment (jfly) [reply WcaOnRails/app/views/static_pages/teams_committees.html.erb]: I just merged up https://github.com/thewca/worldcubeassociation.org/pull/3786, which removes `all_ordered_by_english_name.each` and replaces it with `all_official`, which will *not* include the Board, so hopefully that cleans this up a little bit when you rebase.

**Comment by AlbertoPdRF** [comment]
@UnsolvedCypher please see #3808 as it adds the Officers Teams.

**Comment by cubewhiz** [comment]
Any progress lately?

**Comment by UnsolvedCypher** [comment]
Ok, this PR is in better shape now as I've rebased and rewritten the officer roles as per #3808 and also changed some wording to be appropriate for Councils. However, I'm having some trouble understanding why these tests are failing. Is it just because a part of the navigation is untranslated?

**Comment by viroulep** [comment]
Indeed the first 3 errors occurred because we have a couple of tests which rely on i18n, and basically we do need a couple of translation when changing the navigation bar.
For French you can add:
```yaml
teams: "Équipes, Comités, et Conseils"
```
Somewhere below [here](https://github.com/thewca/worldcubeassociation.org/blob/c12b5d2dbb611a977905cf37b80bdf93f82e9c12/WcaOnRails/config/locales/fr.yml#L1076).

@AlbertoPdRF or @moralsh can probably give input on what the appropriate `es` translation would be!

The last failure is a random test failure we sometimes have, we believe it's due to us loading gmaps in tests, but currently our workaround is to restart the build :s

**Comment by UnsolvedCypher** [comment]
Hello all, this PR is ready to be reviewed again.

- @jonatanklosko thank you for your detailed review, I have implemented all of these changes
- I internationalized some parts that were not internationalized previously. I think that should be all, but if you find anything that needs i18n, please let me know
- @viroulep thank you for your explanation, I've added Spanish and French keys and the tests passed
- I have modified the mobile layout to force one person per row as @cubewhiz suggested, which you can see in the below screenshot. I think this looks good, but I am open to other ideas
- I've changed the officers to have their labels automatically created by reading the membership of the officer teams, so these aren't hardcoded anymore
- I have added "Council" to any places that previously didn't reference it and now need to after the introduction of the WAC. Councils will show up below any Teams and Committees, but this behavior can be changed if the consensus is that they should be alphabetized like the rest of the teams.

Below are the screenshots of the updated pages:

Desktop screenshot:
Desktop screenshot

Mobile screenshot:
Mobile screenshot

Please let me know if you have any feedback either about the design or the code.

File: WcaOnRails/app/views/static_pages/teams_committees.html.erb [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+  <br />
+
+  <% @officers.each do |user, positions| %>
+    <div class="badge team-member-badge officer-badge">
+        <%= render "shared/user_avatar", user: user %>
+        <div class="team-member-name"><%= user.name %>
+          <br />
+          <span class="team-leader-subtext"><%= positions %></span>
+        </div>
+      </div>
```
Comment (jonatanklosko) [inline WcaOnRails/app/views/static_pages/teams_committees.html.erb]: Nit: indentation is a bit off here.

**Comment by UnsolvedCypher** [comment]
Thanks for the review @jonatanklosko! I fixed the indentation so the PR should be set now. Please do not merge this yet as we are still waiting for final approval of the Board and any feedback from other Teams and Committees.

**Comment by UnsolvedCypher** [comment]
Ok, looks like the tests failed again due to the Google Maps issue. Could a WST member please trigger a Travis rebuild?

**Comment by viroulep** [comment]
Note: the "worlds-2017" image is actually taken from WC 2015 :)

**Comment by UnsolvedCypher** [comment]
@viroulep updated, thanks!

**Comment by AlbertoPdRF** [comment]
Chris is missing its "WCA Treasurer" role. As our Motions define the Treasurer as the WFC Leader it was decided to not create a redundant officer team for it.

Also, maybe it would look nicer if different officer roles where on different lines?

Apart from that it looks good to me, thanks for the great effort!

**Comment by UnsolvedCypher** [comment]
@AlbertoPdRF thank you for the suggestion! I've added the treasurer to the list of titles, and then had to do some CSS changes to make sure the officer boxes would be the same size.

You can see an exaggerated version of what this looks like when some officers have significantly more titles than others below. It definitely looks better in the more realistic case where officers have around the same number of titles, though.

Screenshot from 2019-03-25 18-06-08

**Comment by cubewhiz** [comment]
I don't think you need Board approval for this--all Board members have had ample time to review this PR by now.

File: WcaOnRails/app/assets/stylesheets/teams-and-committees.scss [hunk 6]
```diff
@@ -0,0 +1,36 @@
...
+  background-color: black;
+  color: white;
+}
+
+.badge.team-email {
+  background-color: #e64503;
+  a {
+    color: white;
+  }
+}
```
[the remark this answers is not in the document]
Comment (pedrosino) [reply WcaOnRails/app/assets/stylesheets/teams-and-committees.scss]: Seems like there's no common property for .badge alone. Is there a reason to group them?

File: WcaOnRails/app/controllers/static_pages_controller.rb [hunk 7]
```diff
@@ -159,6 +159,21 @@ def score_tools
   def logo
   end
 
+  def teams_committees
+    # get all users who hold one or more officer positions
+    officer_users = Team.all_officers.map(&:current_members).inject(&:+).map(&:user)
+    treasurers = Team.wfc.current_members.filter(&:team_leader).map(&:user)
```
Comment (pedrosino) [inline WcaOnRails/app/controllers/static_pages_controller.rb]: This line is failing at stagin with

410374b]   CACHE TeamMember Load (0.0ms)  SELECT  `team_members`.* FROM `team_members` WHERE `team_members`.`team_id` = 6 AND (end_date IS NULL OR end_date > '2019-04-20') LIMIT 11
F, [2019-04-20T00:02:26.322810 #13050] FATAL -- : [7e2e1829-396c-486c-a7ea-1a591410374b] NoMethodError (undefined method `filter' for #<TeamMember::ActiveRecord_Associations_CollectionProxy:0x000000000936bd80>):
F, [2019-04-20T00:02:26.322838 #13050] FATAL -- : [7e2e1829-396c-486c-a7ea-1a591410374b]   
F, [2019-04-20T00:02:26.322858 #13050] FATAL -- : [7e2e1829-396c-486c-a7ea-1a591410374b] app/controllers/static_pages_controller.rb:165:in `teams_committees'
[7e2e1829-396c-486c-a7ea-1a591410374b] lib/middlewares/warden_user_logger.rb:13:in `call'
[7e2e1829-396c-486c-a7ea-1a591410374b] lib/middlewares/fix_accept_header.rb:14:in `call'

Comment (jonatanklosko) [reply WcaOnRails/app/controllers/static_pages_controller.rb]: Indeed, it should be `select` =)

File: WcaOnRails/app/assets/stylesheets/teams-and-committees.scss (hunk 6 above)
[the remark this answers is not in the document]
Comment (UnsolvedCypher) [reply WcaOnRails/app/assets/stylesheets/teams-and-committees.scss]: Badge is a [Bootstrap component](https://getbootstrap.com/docs/4.3/components/badge/) which is why they all have the badge class. The grouping is just for organizational purposes, but this can be changed if you think it's less clean.

[the remark this answers is not in the document]
Comment (pedrosino) [reply WcaOnRails/app/assets/stylesheets/teams-and-committees.scss]: That's fine.

File: WcaOnRails/app/controllers/static_pages_controller.rb (hunk 7 above)
Comment (pedrosino) [reply WcaOnRails/app/controllers/static_pages_controller.rb]: @UnsolvedCypher Could you fix this? I think it's the last thing, apart from rebasing.

Comment (UnsolvedCypher) [reply WcaOnRails/app/controllers/static_pages_controller.rb]: @pedrosino yes, this is all set now :)

Comment (pedrosino) [reply WcaOnRails/app/controllers/static_pages_controller.rb]: Actually there are two other occurrences just below, which are still failing.

Comment (UnsolvedCypher) [reply WcaOnRails/app/controllers/static_pages_controller.rb]: Thank you, I just fixed those

File: WcaOnRails/app/views/shared/_user_avatar.html.erb [hunk 8]
```diff
@@ -1,5 +1,6 @@
 <% avatar_symbol = defined?(show_pending) && show_pending ? :pending_avatar : :avatar %>
   <% url = user.send(avatar_symbol).url(:thumb) + ( defined?(break_cache) ? "?#{Time.now.to_f}" : "" ) %>
+  <% url.gsub!("localhost:3000", "worldcubeassociation.org") %>
```
Comment (jonatanklosko) [inline WcaOnRails/app/views/shared/_user_avatar.html.erb]: Hmm, I believe we don't need that, the origin just depends on environment. Why was that change needed?

Comment (UnsolvedCypher) [reply WcaOnRails/app/views/shared/_user_avatar.html.erb]: @jonatanklosko Thank you for catching this, this is workaround I've been using to let the images display locally so I could get a better idea of how everything looked. I had been reverting this before my commits but it looks like I forgot this time.

**Comment by UnsolvedCypher** [comment]
Alright, I rebased but I'm getting the following errors: 


```
Failed examples:

[31mrspec ./spec/controllers/competitions_controller_spec.rb:184[0m [36m# CompetitionsController GET #for_senior when signed in as a regular Delegate does not allow access[0m
[31mrspec ./spec/controllers/competitions_controller_spec.rb:175[0m [36m# CompetitionsController GET #for_senior when signed in as a senior Delegate renders the for_senior page[0m
[31mrspec ./spec/features/competition_events_spec.rb:86[0m [36m# Competition events management unconfirmed competition change round attributes change cutoff to best of 2 in 2 minutes[0m
```

I'm a bit confused since none of this should have anything to do with changes I've been working on.

File: WcaOnRails/config/routes.rb [hunk 9]
```diff
@@ -33,7 +33,6 @@
   get 'map' => 'competitions#embedable_map'
 
   get 'competitions/mine' => 'competitions#my_competitions', as: :my_comps
-  get 'competitions/for_senior(/:user_id)' => 'competitions#for_senior', as: :competitions_for_senior
```
Comment (pedrosino) [inline WcaOnRails/config/routes.rb]: Hey @UnsolvedCypher 
Here is the problem of your tests failing. You removed this row.

Comment (UnsolvedCypher) [reply WcaOnRails/config/routes.rb]: Thank you for catching that and sorry for my delay in response. I added this back in and tests are passing now. Please let me know if there are any other changes you would like.

**Comment by AlbertoPdRF** [comment]
Is there anything left for this PR to be ready?

**Comment by pedrosino** [comment]
@UnsolvedCypher  will rebase again so hopefully it will let me merge.

**Comment by pedrosino** [comment]
Closed in favor of #4269
